src/commands/synthetics/__tests__/run-test.test.ts

```diff
@@ -35,23 +33,24 @@ describe('run-test', () => {
       const configOverride = {locations, startUrl}
 
       const apiHelper = {}
-      const command = new RunTestCommand()
-      command.context = {stdout: {write: jest.fn()}} as any
-      command['getApiHelper'] = (() => apiHelper) as any
-      // Override with config file
-      jest
-        .spyOn(ciUtils, 'getConfig')
-        .mockImplementation(async () => ({global: configOverride, publicIds: ['public-id-1', 'public-id-2']}))
-      await command.execute()
-
-      expect(getTestsToTriggersMock).toHaveBeenCalledWith(
-        apiHelper,
-        expect.arrayContaining([
-          expect.objectContaining({id: 'public-id-1', config: configOverride}),
-          expect.objectContaining({id: 'public-id-2', config: configOverride}),
-        ]),
-        expect.anything()
-      )
+
+      jest.spyOn(runTests, 'getApiHelper').mockImplementation(() => ({} as any))
+      try {
+        await runTests.executeTests(mockReporter, {
+          ...config,
+          global: configOverride,
+          publicIds: ['public-id-1', 'public-id-2'],
+        })
+      } catch (error) {
+        expect(getTestsToTriggersMock).toHaveBeenCalledWith(
+          apiHelper,
+          expect.arrayContaining([
+            expect.objectContaining({id: 'public-id-1', config: configOverride}),
+            expect.objectContaining({id: 'public-id-2', config: configOverride}),
+          ]),
+          expect.anything()
+        )
+      }
```

Comment:
I am not sure to understand why you wrap `executeTests` in a `try ... catch`.
Either you expect it to not throw, and simply remove it. Or you expect it to throw and you can write:
```typescript
expect(runTests.executeTests(...)).rejects.toThrow(SomeError)
```

If you want, let's take a moment to clarify how `expect` and `test` work in jest :)

Reply:
Done!
Following our off-line conversation, I've added a matcher in order to assert on the error's code and instance type :
```
expect.extend({
  toBeError(received, errorClass, errorCode) {
    const pass = received instanceof errorClass && errorCode === received.code
    if (pass) {
      return {
        message: () =>
          `expected ${received.constructor.name} with code ${received.code} but received ${errorClass.name} with code ${errorCode}`,
        pass: true,
      }
    } else {
      return {
        message: () =>
          `Received ${received.constructor.name} with code ${received.code} but expected ${errorClass.name} Error with code ${errorCode}`,
        pass: false,
      }
    }
  },
})
```

Reply:
As discussed offline, have resorted to using this package instead : https://github.com/DanielHreben/jest-matcher-specific-error